sources/constraint-declaration-validation.asciidoc

```diff
@@ -166,6 +165,22 @@ public class User {
 ----
 ====
 
+.Making a nested collection subject to graph validation
+====
+[source, JAVA]
+----
+public class User {
+    private Map<String, @Valid List<@Valid Address>> addressesByType;
```

Comment:
That's the question essentially; I've filed [BVAL-601](https://hibernate.atlassian.net/browse/BVAL-601) to clarify it (last question in there). For now I'd prefer to stay safe and require them all along the path.